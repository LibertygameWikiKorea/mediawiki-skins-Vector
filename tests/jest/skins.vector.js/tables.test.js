const tables = require( '../../../resources/skins.vector.js/tables.js' ).init;

describe( 'tables', () => {
	test( 'wraps table with div', () => {
		document.body.innerHTML = `
			<section class="mw-parser-output">
				<table>
					<tbody><tr><th>table table table</th></tr></tbody>
				</table>
			</section>
		`;
		tables();

		expect( document.body.innerHTML ).toMatchSnapshot();
	} );

	test( 'wraps multiple table with div', () => {
		document.body.innerHTML = `
			<section class="mw-parser-output">
				<table>
					<tbody><tr><th>table table table</th></tr></tbody>
				</table>
				<table>
					<tbody><tr><th>table table table</th></tr></tbody>
				</table>
			</section>
		`;
		tables();

		expect( document.body.innerHTML ).toMatchSnapshot();
	} );

	test( 'doesnt wrap nested tables', () => {
		document.body.innerHTML = `
			<section class="mw-parser-output">
				<table>
					<tbody>
						<tr><th>table table table</th></tr>
						<tr><td><table><tbody><tr><th>table table table</th></tr></tbody></table><td></tr>
					</tbody>
				</table>

			</section>
		`;
		tables();

		expect( document.body.innerHTML ).toMatchSnapshot();
	} );
} );
