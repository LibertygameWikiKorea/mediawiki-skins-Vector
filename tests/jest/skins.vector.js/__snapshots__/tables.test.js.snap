// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tables doesnt wrap nested tables 1`] = `
"
			<section class=\\"mw-parser-output\\">
				<div class=\\"noresize\\"><table>
					<tbody>
						<tr><th>table table table</th></tr>
						<tr><td><table><tbody><tr><th>table table table</th></tr></tbody></table></td><td></td></tr>
					</tbody>
				</table></div>

			</section>
		"
`;

exports[`tables wraps multiple table with div 1`] = `
"
			<section class=\\"mw-parser-output\\">
				<div class=\\"noresize\\"><table>
					<tbody><tr><th>table table table</th></tr></tbody>
				</table></div>
				<div class=\\"noresize\\"><table>
					<tbody><tr><th>table table table</th></tr></tbody>
				</table></div>
			</section>
		"
`;

exports[`tables wraps table with div 1`] = `
"
			<section class=\\"mw-parser-output\\">
				<div class=\\"noresize\\"><table>
					<tbody><tr><th>table table table</th></tr></tbody>
				</table></div>
			</section>
		"
`;
